crates/typst-syntax/src/lexer.rs

```diff
@@ -121,6 +124,17 @@ impl Lexer<'_> {
         }
     }
 
+    pub fn next_non_trivia(&mut self) -> SyntaxKind {
```

Comment:
I wonder whether this method should really be on the lexer or rather be specific to the parser implementation. The reasoning is that this method is easy to misuse: It should only ever be called on a cloned lexer since it will otherwise discard tokens, messing up the syntax tree.

Reply:
I was torn tbh.. but the reason of abuse-proof here makes sense!